### PR TITLE
PLT-18: Create validator/minting policy/stake validator wrapper for Plutus V2

### PR DESCRIPTION
**Pull request by koslambrou**
The main changes are done in the packages `plutus-script-utils`. I created a `V2` module from the `V1`, then I updated the functions to work with Plutus V2.

Unfortunately, I also did a merge from `main` branch, so it also includes changes from commits from main.

So, mainly look at the changes `plutus-script-utils`.

Pre-submit checklist:
- Branch
    - [x] Tests are provided (if possible)
    - [x] Commit sequence broadly makes sense
    - [x] Key commits have useful messages
    - [x] Relevant tickets are mentioned in commit messages
    - [x] Formatting, materialized Nix files, PNG optimization, etc. are updated
- PR
    - [x] Self-reviewed the diff
    - [x] Useful pull request description
    - [x] Reviewer requested

### CODE REVIEW

File: plutus-script-utils/src/Plutus/Script/Utils/V1/Scripts.hs [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+from `cardano-ledger`.
+The good: the lowest dependency footprint.
+The bad: code duplication.
+
+However, we expect that most Plutus script devs depending on this package will
+also probably depend on `cardano-api`, so the dependency on `cardano-api` should
+(probably) be an non-issue.
+
+If this becomes an issue, we'll change the implementation.
+-}
```
Comment (ak3n) [inline plutus-script-utils/src/Plutus/Script/Utils/V1/Scripts.hs:132]: 👍

File: plutus-script-utils/src/Plutus/Script/Utils/V1/Typed/Scripts/Validators.hs [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+{-# LANGUAGE DerivingStrategies  #-}
+{-# LANGUAGE TypeFamilies        #-}
+
+{-# OPTIONS_GHC -fno-specialise #-}
+{-# OPTIONS_GHC -Wno-simplifiable-class-constraints #-}
+{-# OPTIONS_GHC -fno-omit-interface-pragmas #-}
+
+module Plutus.Script.Utils.V1.Typed.Scripts.Validators
+    ( UntypedValidator
+    , mkUntypedValidator
```
Comment (ak3n) [inline plutus-script-utils/src/Plutus/Script/Utils/V1/Typed/Scripts/Validators.hs:12]: Isn't it strange that we provide `Untyped` things from the `Typed` submodules? Shouldn't they belong to `Untyped`?

File: plutus-script-utils/src/Plutus/Script/Utils/V2/Scripts.hs [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    -- * Script utilities
+    , scriptCurrencySymbol
+    ) where
+
+import Cardano.Api qualified as Script
+import Cardano.Api.Shelley qualified as Script
+import Codec.Serialise (serialise)
+import Data.ByteString.Lazy qualified as BSL
+import Data.ByteString.Short qualified as SBS
+import Plutus.Script.Utils.V1.Scripts qualified as P
```
Comment (ak3n) [inline plutus-script-utils/src/Plutus/Script/Utils/V2/Scripts.hs:30]: Better to be consistent and `PV1`?

Sorry for being too pedantic.

**Review by ak3n — APPROVED** [review]

File: plutus-script-utils/src/Plutus/Script/Utils/V2/Scripts.hs (hunk 3 above)
Comment (koslambrou) [reply plutus-script-utils/src/Plutus/Script/Utils/V2/Scripts.hs:30]: Not at all :) 

However, where does the `PV1` come from? `plutus-ledger-api` defines `V1`, not `PV1`.

Comment (ak3n) [reply plutus-script-utils/src/Plutus/Script/Utils/V2/Scripts.hs:30]: In other modules the imports are usually end with `qualified as PV1/PV2`.

File: plutus-script-utils/src/Plutus/Script/Utils/V1/Typed/Scripts/Validators.hs (hunk 2 above)
Comment (koslambrou) [reply plutus-script-utils/src/Plutus/Script/Utils/V1/Typed/Scripts/Validators.hs:12]: Mmm... good point. I guess we could put it in `Plutus.Script.Utils.V1.Scripts`.

Comment (koslambrou) [reply plutus-script-utils/src/Plutus/Script/Utils/V1/Typed/Scripts/Validators.hs]: Can you confirm that the new modules make sense?

Comment (ak3n) [reply plutus-script-utils/src/Plutus/Script/Utils/V1/Typed/Scripts/Validators.hs]: Yeah, looks good 👍

**Pull request merged**